<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ColorPaletteEdit</class>
 <widget class="QDialog" name="ColorPaletteEdit">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>341</width>
    <height>613</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>编辑调色板</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>调色板</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="maxTitle">
        <property name="text">
         <string>上限</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="max">
        <property name="minimum">
         <double>-999999.989999999990687</double>
        </property>
        <property name="maximum">
         <double>999999.989999999990687</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="minTitle">
        <property name="text">
         <string>下限</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="min">
        <property name="minimum">
         <double>-999999.989999999990687</double>
        </property>
        <property name="maximum">
         <double>999999.989999999990687</double>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="emptyColorTitle">
        <property name="text">
         <string>无数据</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>类型</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QRadioButton" name="interpolationLinear">
           <property name="text">
            <string>线性渐变</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="interpolationStep">
           <property name="text">
            <string>阶梯</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <layout class="QGridLayout" name="gridLayout">
        <property name="horizontalSpacing">
         <number>0</number>
        </property>
        <item row="1" column="0" colspan="3">
         <widget class="QWidget" name="arrows" native="true">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <spacer name="arrowSpacerLeft">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>25</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="2">
         <spacer name="arrowSpacerRight">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>25</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1">
         <widget class="ColorPaletteView" name="gradient" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="7" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QToolButton" name="prev">
          <property name="toolTip">
           <string>选择上一个颜色</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="arrowType">
           <enum>Qt::LeftArrow</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="next">
          <property name="toolTip">
           <string>选择下一个颜色</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="arrowType">
           <enum>Qt::RightArrow</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="addStop">
          <property name="toolTip">
           <string>添加颜色</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/Resources.qrc">
            <normaloff>:/icons/plus.png</normaloff>:/icons/plus.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="removeStop">
          <property name="toolTip">
           <string>删除颜色</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/Resources.qrc">
            <normaloff>:/icons/minus.png</normaloff>:/icons/minus.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="tools">
          <property name="toolTip">
           <string>其他选项</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/icons/Fatcow.qrc">
            <normaloff>:/fatcow/16x16/cog.png</normaloff>:/fatcow/16x16/cog.png</iconset>
          </property>
          <property name="popupMode">
           <enum>QToolButton::InstantPopup</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="autoMax">
        <property name="text">
         <string>自动上限</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="autoMin">
        <property name="text">
         <string>自动下限</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="Utils::QtColorButton" name="emptyColor"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="editColor">
     <property name="title">
      <string>编辑颜色</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="4" column="1">
       <widget class="QSpinBox" name="greenValue">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QSlider" name="blue">
        <property name="styleSheet">
         <string notr="true">::handle {background-color: rgb(0, 0, 255);}</string>
        </property>
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <layout class="QFormLayout" name="formLayout_2">
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="position">
          <property name="minimum">
           <double>-999999.989999999990687</double>
          </property>
          <property name="maximum">
           <double>999999.989999999990687</double>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="Utils::QtColorButton" name="color"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="colorTitle">
          <property name="text">
           <string>选择颜色</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="positionTitle">
          <property name="text">
           <string>位置</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QSlider" name="red">
        <property name="styleSheet">
         <string notr="true">::handle {background-color: rgb(255, 0, 0);}</string>
        </property>
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QSlider" name="green">
        <property name="styleSheet">
         <string notr="true">::handle {background-color: rgb(0, 255, 0);}</string>
        </property>
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QSpinBox" name="blueValue">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QSpinBox" name="alphaValue">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QSlider" name="alpha">
        <property name="styleSheet">
         <string notr="true">::handler {background-image: url(:/TransparentBackground.png);}</string>
        </property>
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="redValue">
        <property name="maximum">
         <number>255</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok|QDialogButtonBox::RestoreDefaults</set>
     </property>
    </widget>
   </item>
  </layout>
  <action name="actionImport">
   <property name="icon">
    <iconset resource="../../resources/icons/Fatcow.qrc">
     <normaloff>:/fatcow/16x16/application_go.png</normaloff>:/fatcow/16x16/application_go.png</iconset>
   </property>
   <property name="text">
    <string>导入</string>
   </property>
   <property name="toolTip">
    <string>从文件导入一个调色板</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="icon">
    <iconset resource="../../resources/icons/Fatcow.qrc">
     <normaloff>:/fatcow/16x16/application_get.png</normaloff>:/fatcow/16x16/application_get.png</iconset>
   </property>
   <property name="text">
    <string>导出</string>
   </property>
   <property name="toolTip">
    <string>将调色板导出为文件</string>
   </property>
  </action>
  <action name="actionAutoSpace">
   <property name="icon">
    <iconset resource="../../resources/icons/Fatcow.qrc">
     <normaloff>:/fatcow/16x16/size_horizontal.png</normaloff>:/fatcow/16x16/size_horizontal.png</iconset>
   </property>
   <property name="text">
    <string>均匀分布</string>
   </property>
   <property name="toolTip">
    <string>调整各个颜色点的位置, 使其均匀分布</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorPaletteView</class>
   <extends>QWidget</extends>
   <header>Imaging/ColorPaletteView.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Utils::QtColorButton</class>
   <extends>QToolButton</extends>
   <header>utils/qtcolorbutton.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../resources/icons/Fatcow.qrc"/>
  <include location="../../resources/Resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ColorPaletteEdit</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>533</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ColorPaletteEdit</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>533</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>red</sender>
   <signal>valueChanged(int)</signal>
   <receiver>redValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>398</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>green</sender>
   <signal>valueChanged(int)</signal>
   <receiver>greenValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>425</y>
    </hint>
    <hint type="destinationlabel">
     <x>298</x>
     <y>428</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>blueValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>198</x>
     <y>451</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>454</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>alpha</sender>
   <signal>valueChanged(int)</signal>
   <receiver>alphaValue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>483</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>red</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>297</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>135</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>greenValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>green</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>424</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>428</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blueValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>blue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>294</x>
     <y>454</y>
    </hint>
    <hint type="destinationlabel">
     <x>160</x>
     <y>455</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>alphaValue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>alpha</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>297</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>159</x>
     <y>480</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
